modin/pandas/base.py

```diff
@@ -245,47 +245,42 @@ def _validate_other(
         TypeError
             If any validation checks fail.
         """
-        # We skip dtype checking if the other is a scalar.
-        if is_scalar(other):
+        if isinstance(other, BasePandasDataset):
+            return other._query_compiler
```

Comment:
Returns sections of the docstring should be corrected.

Reply:
Do you say that because the function can return query compiler? That's true, but that problem precedes this PR. See deleted lines 253-254 in this file.

Reply:
I see, but please file an issue to fix this.

Reply:
filed #5282